### PR TITLE
[FEATURE][processing][needs-docs]Script search

### PR DESCRIPTION
**Pull request by volaya**
## Description

Added search and replace functionality to script editors in Processing

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: python/plugins/processing/gui/BatchPanel.py [hunk 1]
```diff
@@ -178,9 +179,15 @@ def getWidgetFromParameter(self, param, row, col):
 
         return item
 
+    def getLastFolder():
+        settings = QSettings()
```
Comment (nyalldawson) [inline python/plugins/processing/gui/BatchPanel.py]: QgsSettings

**Review by nyalldawson — COMMENTED** [review]
Looks good to me!

File: python/plugins/processing/gui/BatchPanel.py (hunk 1 above)
Comment (volaya) [reply python/plugins/processing/gui/BatchPanel.py]: yikes, that's from a different commit that adds the ability to remember the last folder for batch processes. Looks like i mixed things up. Let me re push this PR :-)

Comment (volaya) [reply python/plugins/processing/gui/BatchPanel.py]: Should be fine now

**Comment by Gustry** [comment]
Is-it a new small feature in 2.18?

**Comment by volaya** [comment]
yes, i think there was a ticket for it. I did this some time ago, but never commited it. Should be easy to port to master as well.

**Comment by alexbruy** [comment]
Thanks! Will forward port to master too.

**Comment by anitagraser** [comment]
Thanks! For reference, the ticket was https://issues.qgis.org/issues/15646

**Pull request merged**